nova/virt/vmwareapi/vmops.py

```diff
@@ -2502,18 +2536,22 @@ def list_instances(self):
 
         return lst_vm_names
 
-    def _list_instances_in_cluster(self):
+    def _list_instances_in_cluster(self, additional_properties=None):
```

Comment:
since .extend is used, the additional_properties is expected to be a [], right?
(suppling string will result in each single character added to properties)

Reply:
Given that the name of the variable is a plural, I would expect callers of that function to pick up on the fact that they should pass an iterable containing one or multiple values. Do you expect the code to type-check that argument? Because we don't usually do that, even if we expect lists.

Reply:
Nope, I don't think that type check is needed.
for iterables I would usually define _additional_properties=[]_ as a default and do check as _if additional_properties:_
but it is up to you

Reply:
The problem with having a list-instance as default-argument is, that once you start adding to that list, you changed the default-argument. That wouldn't be a problem here, but that's why I refrain from using list-instances as default-arguments.